tldw_Server_API/app/api/v1/endpoints/character_chat_sessions.py

```
@@ -236,6 +259,281 @@ def _safe_replace_placeholders(value: Any, char_name: str, user_name: str) -> st
+        if not provider_name:
+            continue
+        preferred_models = get_override_model_priority(provider_name, objective=objective)
+        if preferred_models:
+            provider_order[provider_name] = preferred_models
+    return provider_order
+
+
+async def _select_auto_character_llm_router_choice(
+    *,
+    router_request: RouterRequest,
+    policy: Any,
```

Ruff (0.15.6)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `policy`

(ANN401)